src/core/utils/addons.js

```diff
@@ -111,3 +117,33 @@ export const getAddonJsonLinkedData = ({
     aggregateRating,
   });
 };
+
+export const getPromotedCategory = ({
+  addon,
+  clientApp,
+  forBadging = false,
+}: {|
+  addon: AddonType | CollectionAddonType | SuggestionType | null | void,
+  clientApp: string,
+  forBadging?: boolean,
+|}): PromotedCategoryType | null => {
+  let category = null;
+  if (addon && addon.promoted && addon.promoted.apps.includes(clientApp)) {
+    category = addon.promoted.category;
+  }
+
+  // Special logic if we're using the category for badging.
+  if (forBadging) {
+    // SPONSORED is badged as VERIFIED.
+    if (category === SPONSORED) {
+      category = VERIFIED;
+    }
+
+    // We only have badges for LINE, RECOMMENDED and VERIFIED.
+    if (![LINE, RECOMMENDED, VERIFIED].includes(category)) {
```

Comment:
can we re-use `PromotedBadgeCategory` in src/ui/components/IconPromotedBadge/index.js so we're not defined this logic twice?

Reply:
`PromotedBadgeCategory` is a `type`, so it's only used by Flow for type safety, but it's a good idea to create a constant for this and use that. I'll do that.